applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,368 +57,300 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
+     * <p>
+     * The label value for the ConfigurationSetting is optional. If not specified, the
+     * {@link ConfigurationSetting#NULL_LABEL} is used.
+     * </p>
+     *
+     * If the {@link ConfigurationSetting#etag()} is specified, the configuration value is only updated if it matches.
+     *
+     * @param configurationSetting The key, value, and optional label to set.
+     * @return ConfigurationSetting that was updated.
+     * @throws com.azure.common.http.rest.RestException When a ConfigurationSetting with the same key and label does not
+     *                                               exists or the configuration value is locked.
      */
-    public Mono<RestResponse<ConfigurationSetting>> deleteKeyValue(String key) {
-        return deleteKeyValue(key, null, null);
+    public Mono<RestResponse<ConfigurationSetting>> updateKeyValue(ConfigurationSetting configurationSetting) {
+        Validator.validate(configurationSetting);
+        KeyValueCreateUpdateParameters parameters = new KeyValueCreateUpdateParameters()
+                .withValue(configurationSetting.value())
+                .withContentType(configurationSetting.contentType())
+                .withTags(configurationSetting.tags());
+
+        String etag = configurationSetting.etag() == null ? ETAG_ANY : configurationSetting.etag();
+
+        return service.setKey(baseUri.toString(), configurationSetting.key(), configurationSetting.label(), parameters, getETagValue(etag), null);
     }
 
+
     /**
-     * Deletes the ConfigurationSetting.
-     * @param key key of the keyValue to delete
-     * @param filter eTag filter to add to If-Match or If-None-Match header
-     * @return the deleted ConfigurationSetting or none if didn't exist.
+     * Gets a ConfigurationSetting that matches the {@param key} and {@param label}.
+     *
+     * @param key The key being retrieved
+     * @return The configuration value in the service.
+     * @throws com.azure.common.http.rest.RestException with status code of 404 if the {@param key} and {@param label} does
+     *                                               not exist.
      */
-    public Mono<RestResponse<ConfigurationSetting>> deleteKeyValue(String key, String label, ETagFilter filter) {
-        if (key == null || key.isEmpty()) {
-            throw new IllegalArgumentException("Parameter key is required and cannot be null or empty");
-        }
-        if (filter != null) {
-            return service.delete(baseUri.toString(), key, label, filter.ifMatch(), filter.ifNoneMatch());
-        }
-        return service.delete(baseUri.toString(), key, label, null, null);
+    public Mono<RestResponse<ConfigurationSetting>> getKeyValue(String key) {
+        return getKeyValue(key, null, null);
     }
 
     /**
-     * Lists the KeyValues.
-     * @param filter query options
-     * @return KeyValues
+     * Gets the ConfigurationSetting given the {@param key}, optional {@param label} and optional {@param etag}.
```

Comment:
For Python and Java we are not providing `etag` parameter as this scenario is not one that we expect a bunch of customers to hit.

Reply:
See the latest updates on https://microsoft.sharepoint.com/teams/AzureDeveloperExperience/_layouts/OneNote.aspx?id=%2Fteams%2FAzureDeveloperExperience%2FShared%20Documents%2FService%20-%20Azconfig%2FAzure%20SDK%20-%20Service%20-%20Azconfig&wd=target%28Azconfig.one%7C50856656-5D71-2640-B240-11758DB2FB6E%2FStandarization%20across%20languages%7C79180E7A-F7B4-4221-8398-D440467E3782%2F%29
onenote:https://microsoft.sharepoint.com/teams/AzureDeveloperExperience/Shared%20Documents/Service%20-%20Azconfig/Azure%20SDK%20-%20Service%20-%20Azconfig/Azconfig.one#Standarization%20across%20languages%20&section-id={50856656-5D71-2640-B240-11758DB2FB6E}&page-id={79180E7A-F7B4-4221-8398-D440467E3782}&object-id={AD444C44-9B38-0225-20E9-23FDAB509498}&22